src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ConditionalWcfTest.cs

```diff
@@ -10,42 +15,191 @@ namespace Infrastructure.Common
+                                                           {
+                                                               return false;
+                                                           }
+                                                       });
         }
 
         public static bool SPN_Available()
         {
-            return IsConditionalTestEnabled(TestProperties.SPN_Available_PropertyName);
+            // If TestProperties does not specify true/false explicitly, we need to autodetect whether
+            // the SPN is available.
+            // Temporarily use the simple heuristic that if we are running the services locally, it is.
+            // Refactor this after integration to address https://github.com/dotnet/wcf/issues/1024 
+            return GetConditionValueFromTestProperties(TestProperties.SPN_Available_PropertyName,
+                                                        Detect_Server_Is_Local);
         }
 
         public static bool Kerberos_Available()
         {
-            return IsConditionalTestEnabled(TestProperties.Kerberos_Available_PropertyName);
+            // If TestProperties does not specify true/false explicitly, we need to autodetect whether
+            // Kerberos is available.
+            // Temporarily use the simple heuristic that if we are running the services locally, it is.
+            // Refactor this after integration to address https://github.com/dotnet/wcf/issues/1024 
+            return GetConditionValueFromTestProperties(TestProperties.Kerberos_Available_PropertyName,
+                                                        Detect_Server_Is_Local);
+        }
+
+        public static bool Windows_Authorization_Available()
+        {
+            // If TestProperties does not specify true/false explicitly, we need to autodetect whether
+            // Windows Authorization is available.
+            // Temporarily use the simple heuristic that if we are running the services locally, it is.
+            // Refactor this after integration to address https://github.com/dotnet/wcf/issues/1024 
+            return GetConditionValueFromTestProperties(TestProperties.Windows_Authorization_Available_PropertyName,
+                                                        Detect_Server_Is_Local);
         }
 
+        public static bool NTLM_Available()
+        {
+            // If TestProperties does not specify true/false explicitly, we need to autodetect whether
+            // NTLM is available.
+            // Temporarily use the simple heuristic that if we are running the services locally, it is.
+            // Refactor this after integration to address https://github.com/dotnet/wcf/issues/1024 
+            return GetConditionValueFromTestProperties(TestProperties.NTLM_Available_PropertyName,
+                                                        Detect_Server_Is_Local);
+        }
+
+        public static bool WebSockets_Supported()
```

Comment:
I can remove the [ActiveIssue] example to avoid complicating the discussion.  It was intended to foster this very discussion 😀 .

But I think we need to have some clarity between what is a condition and what is an issue. WebSockets on *nix is a good example.  If we mark them only [ActiveIssue], they silently don't run.  If we mark them with [ConditionalFact], then we see "Condition(s) not met: web sockets supported".  My personal opinion is that [ConditionaFact] should be used for these scenarios:

- This test environment is not setup to allow feature X to be tested

- Feature X is not supported in this environment, and never will be

- Feature X is not currently supported in this environment (and an active issue is tracking it)

When there is a legitimate product failure (ex: wrong exception from streaming error) then I think it is a true ActiveIssue, and we need to settle on how that is represented.  We could invent our own [ActiveIssueFact(999)] is this helps distinguish